src/Components/Facility/FacilityDoctorList.tsx

```diff
@@ -0,0 +1,140 @@
+                  Total Doctors
+                </div>
+                <h2 className="mt-2 text-xl font-bold">{totalDoctors}</h2>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        {doctorQuery.data.results.map((data: DoctorModal) => {
+          const removeCurrentDoctorData = (doctorId: number | undefined) => {
+            if (doctorQuery.data !== undefined) {
+              doctorQuery.data.results = doctorQuery.data?.results.filter(
+                (i: DoctorModal) => i.id !== doctorId
+              );
+            }
+          };
+
+          return (
+            <DoctorsCountCard
+              facilityId={props.facilityId}
+              key={`bed_${data.id}`}
+              handleUpdate={async () => {
+                doctorQuery.refetch();
+                if (doctorQuery.res?.ok && doctorQuery.data) {
+                  // update total doctors count
+                  let totalCount = 0;
+                  doctorQuery.data.results.map((doctor: DoctorModal) => {
+                    if (doctor.count) {
+                      totalCount += doctor.count;
+                    }
+                  });
+                  setTotalDoctors(totalCount);
+                }
+              }}
+              {...data}
+              removeDoctor={removeCurrentDoctorData}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div className="mt-5 rounded bg-white p-3 shadow-sm md:p-6">
+        <div className="justify-between md:flex md:pb-2">
+          <div className="mb-2 text-xl font-bold">Doctors List</div>
+          <ButtonV2
+            className="w-full md:w-auto"
+            onClick={() => setDoctorCapacityModalOpen(true)}
+            disabled={doctorList.length === DOCTOR_SPECIALIZATION.length}
+            authorizeFor={NonReadOnlyUsers}
+          >
+            <i className="fas fa-user-md mr-2 text-white" />
+            Add Doctor Types
+          </ButtonV2>
+        </div>
+        <div className="mt-4">{doctorList}</div>
+      </div>
+
+      {doctorCapacityModalOpen && (
+        <DialogModal
+          show={doctorCapacityModalOpen}
+          onClose={() => setDoctorCapacityModalOpen(false)}
+          title="Add Doctor Capacity"
+          className="max-w-md md:min-w-[600px]"
+        >
+          <DoctorCapacity
+            facilityId={props.facilityId}
+            handleClose={() => setDoctorCapacityModalOpen(false)}
+            handleUpdate={async () => {
+              doctorQuery.refetch();
+              if (doctorQuery.res?.ok && doctorQuery.data) {
+                // update total doctors count
+                setTotalDoctors(
+                  doctorQuery.data.results.reduce(
+                    (acc: number, doctor: DoctorModal) =>
+                      acc + (doctor.count || 0),
+                    0
+                  )
+                );
+              }
```

Comment:
Same, total doctors should be treated as side effects of query.data.results instead of having a dedicated state and mutating the state.